gravitee-am-service/src/main/java/io/gravitee/am/service/impl/OrganizationUserServiceImpl.java

```diff
@@ -177,20 +177,24 @@ public Single<AccountAccessToken> generateAccountAccessToken(User user, NewAccou
                     }
                     return accessTokenRepository.create(token)
                             .map(created -> created.toCreateResponse(rawToken))
-                            .flatMap(this::getAccountAccessTokenWithIssuerUsername);
+                            .flatMap(i -> prepareTokenToGet(i, true));
                 });
     }
 
     @Override
     public Flowable<AccountAccessToken> findUserAccessTokens(String organisationId, String userId) {
         return accessTokenRepository.findByUserId(ORGANIZATION, organisationId, userId)
-                .flatMap(token -> getAccountAccessTokenWithIssuerUsername(token).toFlowable());
+                .flatMap(token -> prepareTokenToGet(token).toFlowable());
     }
 
-    private Single<AccountAccessToken> getAccountAccessTokenWithIssuerUsername(AccountAccessToken token) {
-        return token.issuerId() == null ? Single.just(token) :
+    private Single<AccountAccessToken> prepareTokenToGet(AccountAccessToken token) {
+        return prepareTokenToGet(token, false);
+    }
+
+    private Single<AccountAccessToken> prepareTokenToGet(AccountAccessToken token, boolean showToken) {
+        return token.issuerId() == null ? Single.just(token.toBuilder().token(showToken ? token.token() : null).build()) :
                 userRepository.findById(token.issuerId()).defaultIfEmpty(new User())
-                        .map(user -> token.toBuilder().issuerUsername(user.getUsername()).build());
+                        .map(user -> token.toBuilder().token(showToken ? token.token() : null).issuerUsername(user.getUsername()).build());
```

Comment:
Suggestion: you can extract `showToken ? token.token() : null` to variable, since you are using this two times.

Reply:
updated